### PR TITLE
second set of examples with is_sorted, find, for_each - try 2

### PR DESCRIPTION
**Pull request by JeffGarland**
Sorry, don't use git on a daily basis so still learning. Anyway, another set of examples.

### CODE REVIEW

File: example/any_all_none_of.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+int main() {
+
+  std::vector<int> v { 6, 2, 3, 4, 5, 6 };
+  cout << "vector: ";
+  ranges::for_each( v, print ); 
+  cout << "\n";
+  
+  if (ranges::any_of( v, is_six ))   {
+    cout << "vector any_of  is_six: true" << "\n";
```
Comment (CaseyCarter) [inline example/any_all_none_of.cpp]: These might read better as:
```c++
cout << "vector any_of  is_six: " << ranges::any_of( v, is_six ) << '\n';
```
with a `cout << std::boolalpha;` somewhere in the prologue so the result is printed as `true` or `false` and not `1` or `0`. (https://wandbox.org/permlink/bozT2gKCUyjMUqXc)

File: example/find.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  std::vector<int> v { 6, 2, 6, 4, 6, 1 };
+  {
+    auto i = ranges::find( v, 6 ); //1 2 3 4 5 6
+    cout << "*i: " << *i << "\n";
+  }
+  {
+    auto i = ranges::find( v, 10 ); //1 2 3 4 5 6
+    if (i == ranges::end(v) ) { 
+      cout << "didn't find 10" << "\n";
```
Comment (CaseyCarter) [inline example/find.cpp]: `"didn't find 10\n"`

File: example/find.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  cout << "\n" << "array:    ";
+
+  std::array<int, 6> a { 6, 2, 3, 4, 5, 1 };
+  {
+    auto i = ranges::find( a, 6 ); 
+    if (i != ranges::end(a)) {
+      cout << "*i: " << *i;
+    }
+  }
+  cout << "\n" << "list:     ";
```
Comment (CaseyCarter) [inline example/find.cpp]:
```c++
cout << "\nlist:     ";
```

Comment (CaseyCarter) [reply example/find.cpp]: ...and on 90 and 99

File: example/find.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  std::forward_list<int> fl { 6, 2, 3, 4, 5, 1 };
+  {
+    auto i = ranges::find( fl, 4 );
+    if (i != ranges::end(fl)) {
+      cout << "*i: " << *i;
+    }
+  }
+  cout << "\n" << "deque:    ";
+  
+  std::deque<int> d { 6 , 2, 3, 4, 5, 1 };
```
Comment (CaseyCarter) [inline example/find.cpp]: `{ 6 ,` => `{ 6,`

File: example/for_each_assoc.cpp [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+auto print =  [] (int i) { cout << i << " "; };
+//must take a pair for map types
+auto printm = [] (std::pair<string, int> p) { cout << p.first << ":" << p.second << " "; };
+
+int main() {
+  
+  cout  << "set:           ";
+  std::set<int> si { 1, 2, 3, 4, 5, 6 };
+  ranges::for_each( si, print );
+  cout << "\n" << "map:           ";
```
Comment (CaseyCarter) [inline example/for_each_assoc.cpp]: Ditto outputting sequential string literals that could be concatenated. And on 45 and 49.

Comment (CaseyCarter) [reply example/for_each_assoc.cpp]: (and more in the other examples - I'll stop pointing them out now.)

File: example/is_sorted.cpp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+#include <vector>
+#include <array>
+#include <iostream>
+using std::cout;
+
+int main() {
+
+  cout << "vector:   ";
+  std::vector<int> v { 1, 2, 3, 4, 5, 6 };
+  if (ranges::is_sorted( v )) { cout << "true\n"; };
```
Comment (CaseyCarter) [inline example/is_sorted.cpp]: This example is another good case for `std::boolalpha`:
```c++
cout << std::boolalpha;
// ...
cout << ranges::is_sorted( v ) << '\n';
// /...
cout << ranges::is_sorted( a ) << '\n';
```

File: example/find.cpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+#include <range/v3/all.hpp>
+#include <vector>
+#include <deque>
+#include <list>
+#include <forward_list>
+#include <array>
+#include <iostream>
+using std::cout;
+
+auto print = [] (int i) { cout << i << " "; };
```
Comment (CaseyCarter) [inline example/find.cpp]: `print` is unused.

**Comment by ericniebler** [comment]
@JeffGarland can you address the CI problems? And after this, I'd like to prioritize examples that demonstrate how to use the views and actions.

**Comment by JeffGarland** [comment]
Yes I will check the issues. Should be back to this in the next 24 hours.  I will prioritize as you suggest for the next batch.

**Comment by JeffGarland** [comment]
And just checked -- Casey is correct -- unused code is error.  He fixed and I pulled.

**Comment by ericniebler** [comment]
Please also integrate Casey's suggested improvements. It looks like he didn't make those changes himself.

**Comment by ericniebler** [comment]
@JeffGarland Please merge https://github.com/JeffGarland/range-v3/pull/4 into your branch. Then I can merge this.

**Comment by JeffGarland** [comment]
Sorry for the horribly long delay -- restarting my efforts :)

**Comment by ericniebler** [comment]
Welcome back, Jeff. Can you merge JeffGarland/range-v3#5 to fix the build failures on your branch?

**Pull request merged**